e2e/balanceAddress.spec.ts

```diff
@@ -1,20 +1,96 @@
-import { expect, test } from "@playwright/test";
+import { expect, Page, test } from "@playwright/test";
 
+import { satoshiToBtc } from "@/utils/btcConversions";
+import { maxDecimals } from "@/utils/maxDecimals";
+import { trim } from "@/utils/trim";
+
+import {
+  NATIVE_SEGWIT_MAINNET_ADDRESS,
+  NATIVE_SEGWIT_SIGNET_ADDRESS,
+  TAPROOT_MAINNET_ADDRESS,
+  TAPROOT_SIGNET_ADDRESS,
+} from "./constants/wallet";
 import { setupWalletConnection } from "./helper/connect";
+import { mockNetwork } from "./helper/mockNetwork";
+import { extractNumericBalance } from "./helper/utils";
+import { nativeSegwitMainnetBalance } from "./mock/mainnet/wallet/nativeSegwit/utxos";
+import { taprootMainnetBalance } from "./mock/mainnet/wallet/taproot/utxos";
+import { nativeSegwitSignetBalance } from "./mock/signet/wallet/nativeSegwit/utxos";
+import { taprootSignetBalance } from "./mock/signet/wallet/taproot/utxos";
+
+type Network = "mainnet" | "signet";
+type AddressType = "taproot" | "nativeSegwit";
+
+// Determine the network from environment variable or default to 'mainnet'
+const network: Network =
+  (process.env.NEXT_PUBLIC_NETWORK as Network) || "mainnet";
+
+// Define the address types to test
+const addressTypes: AddressType[] = ["taproot", "nativeSegwit"];
+
+// Mappings for expected addresses based on network and address type
+const expectedAddresses: Record<AddressType, Record<Network, string>> = {
+  taproot: {
+    mainnet: TAPROOT_MAINNET_ADDRESS,
+    signet: TAPROOT_SIGNET_ADDRESS,
+  },
+  nativeSegwit: {
+    mainnet: NATIVE_SEGWIT_MAINNET_ADDRESS,
+    signet: NATIVE_SEGWIT_SIGNET_ADDRESS,
+  },
+};
+
+// Mappings for expected balances based on network and address type
+const expectedBalances: Record<AddressType, Record<Network, number>> = {
+  taproot: {
+    mainnet: taprootMainnetBalance,
+    signet: taprootSignetBalance,
+  },
+  nativeSegwit: {
+    mainnet: nativeSegwitMainnetBalance,
+    signet: nativeSegwitSignetBalance,
+  },
+};
```

Comment:
We can move this logic together with current `network` to constants folder and return only taproot/nativeSegwit values depends on current `network` value:

```ts
const addresses = {
  mainnet: {
    taproot: 'address',
    nativeSegwit: 'address',
  },
  signet: {
    taproot: 'address',
    nativeSegwit: 'address',
  }
}[network];
```